QuizApp/QuizApp/LoginViewController.swift

```diff
@@ -2,43 +2,146 @@ import UIKit
+        view.addSubview(scrollView)
+
+        contentView = UIView()
+        scrollView.addSubview(contentView)
+
+        titleLabel = UILabel()
+        contentView.addSubview(titleLabel)
+
         emailInputTextField = CustomInputFieldView(type: .email)
-        view.addSubview(emailInputTextField)
+        contentView.addSubview(emailInputTextField)
 
         passwordInputTextField = CustomInputFieldView(type: .password)
-        view.addSubview(passwordInputTextField)
+        contentView.addSubview(passwordInputTextField)
+
+        loginButton = UIButton()
+        contentView.addSubview(loginButton)
     }
 
     func styleViews() {
-        view.backgroundColor = .purple
+        emailInputTextField.delegate = self
+        passwordInputTextField.delegate = self
+
+        titleLabel.textColor = .white
+        titleLabel.font = .systemFont(ofSize: 32, weight: .bold)
+        titleLabel.textAlignment = .center
+        titleLabel.text = "PopQuiz"
+
+        loginButton.backgroundColor = .white
+        loginButton.setTitleColor(UIColor(red: 0.387, green: 0.16, blue: 0.871, alpha: 1), for: .normal)
+        loginButton.titleLabel?.font = .systemFont(ofSize: 16, weight: .bold)
+        loginButton.setTitle("Login", for: .normal)
+        loginButton.isEnabled = false
+        loginButton.alpha = 0.6
     }
 
     func defineLayoutForViews() {
+        scrollView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+
+        contentView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+            $0.width.equalTo(scrollView.snp.width)
+        }
+
+        titleLabel.snp.makeConstraints {
+            $0.top.equalToSuperview().inset(80)
+            $0.trailing.leading.equalToSuperview().inset(30)
+        }
+
         emailInputTextField.snp.makeConstraints {
-            $0.top.equalToSuperview().inset(140)
+            $0.top.equalTo(titleLabel.snp.bottom).offset(140)
             $0.leading.trailing.equalToSuperview().inset(30)
         }
 
         passwordInputTextField.snp.makeConstraints {
             $0.top.equalTo(emailInputTextField.snp.bottom).offset(18)
             $0.leading.trailing.equalToSuperview().inset(30)
         }
+
+        loginButton.snp.makeConstraints {
+            $0.top.equalTo(passwordInputTextField.snp.bottom).offset(18)
+            $0.leading.trailing.equalToSuperview().inset(30)
+            $0.bottom.equalToSuperview().inset(100)
+            $0.height.equalTo(45)
+        }
+    }
+
+}
+
+extension LoginViewController: CustomInputFieldDelegate {
+
+    func reportChanges(_ type: CustomInputFieldType, _ text: String) {
+        switch type {
+        case .email:
+            email = text
+        case .password:
+            password = text
+        }
+
+        if
+            let email = email,
+            !email.isEmpty,
+            let password = password,
+            !password.isEmpty {
+                loginButton.isEnabled = true
```

Comment:
Please extract styling loginButton in separate functions:
* enableLoginButton
* disableLoginButton